From 260b52e2ca14c6fe87748bcbe260924e0b6e448e Mon Sep 17 00:00:00 2001
From: Samuel Holland <samuel@sholland.org>
Date: Tue, 31 Oct 2023 02:31:47 -0500
Subject: [PATCH 57/68] sunxi: Move SUNXI_GEN_NCAT2 to the board Kconfig

Generation options mostly deal with peripherals, not the CPU
architecture.

Signed-off-by: Samuel Holland <samuel@sholland.org>
---
 arch/arm/mach-sunxi/Kconfig | 8 --------
 board/sunxi/Kconfig         | 6 ++++++
 2 files changed, 6 insertions(+), 8 deletions(-)

--- a/arch/arm/mach-sunxi/Kconfig
+++ b/arch/arm/mach-sunxi/Kconfig
@@ -191,14 +191,6 @@ config SUN50I_GEN_H6
 	Select this for sunxi SoCs which have H6 like peripherals, clocks
 	and memory map.
 
-config SUNXI_GEN_NCAT2
-	bool
-	select MMC_SUNXI_HAS_NEW_MODE
-	select SUPPORT_SPL
-	---help---
-	Select this for sunxi SoCs which have D1 like peripherals, clocks
-	and memory map.
-
 config SUNXI_DRAM_DW
 	bool
 	---help---
--- a/board/sunxi/Kconfig
+++ b/board/sunxi/Kconfig
@@ -80,6 +80,12 @@ config SPL_TEXT_BASE
 	default 0x20060 if SUN50I_GEN_H6 || SUNXI_GEN_NCAT2
 	default 0x00060
 
+config SUNXI_GEN_NCAT2
+	bool
+	help
+	  Select this for sunxi SoCs which have D1-like peripherals,
+	  clocks and memory map.
+
 config SUNXI_MINIMUM_DRAM_MB
 	int
 	default 32 if MACH_SUNIV
